### PR TITLE
cargo-insta: sort snapshots by name

### PR DESCRIPTION
**Pull request by marcospb19**

### CODE REVIEW

File: cargo-insta/src/cli.rs [hunk 1]
```diff
@@ -421,6 +421,7 @@ fn load_snapshot_containers<'a>(
             snapshot_containers.push((snapshot_container?, None));
         }
     }
+    snapshot_containers.sort_by(|(a, _), (b, _)| a.snapshot_path().cmp(&b.snapshot_path()));
```
Comment (marcospb19) [inline cargo-insta/src/cli.rs:424]: This is incredibly helpful for multiple snapshots that come from the same test.

However, it shows them in an odd order:
```
  tests/ui.rs (ui_test_err_missing_files-2)
  tests/ui.rs (ui_test_err_missing_files-3)
  tests/ui.rs (ui_test_err_missing_files-4)
  tests/ui.rs (ui_test_err_missing_files)
```

As a plus, it also groups together snapshots from functions with the same prefix.

**Comment by marcospb19** [comment]
Force-pushing to retrigger tests.

**Comment by mitsuhiko** [comment]
I'm in favor of this change but I would probably add a custom comparison function that detects integer suffixes. If you have more than 10 files it will also sort incorrectly / unexpectedly.

Maybe something like this:

```rust
snapshots.sort_by_key(|x| {
    let mut pieces = x.rsplitn(1, '-');
    if let Some(num_suffix) = pieces.next().and_then(|x| x.parse::<i64>().ok()) {
        (pieces.next().unwrap_or(""), num_suffix)
    } else {
        (*x, 0)
    }
});
```

**Review by mitsuhiko — CHANGES_REQUESTED** [review]
Sorting is reasonable, but it would be better to sort properly in the actual snapshot order.

**Pull request closed without merging**